### PR TITLE
Add contextual menu to bind materials

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
- Adds a bind/unbind contextual menu to the outliner. The menu lists
  available materials to bind.
- Extends the create reference menu to list all asset types that can
  be referenced. Includes MaterialX and Alembic files.

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdContextOps.cpp [hunk 1]
```diff
@@ -684,6 +840,44 @@ Ufe::ContextOps::Items UsdContextOps::getItems(const Ufe::ContextOps::ItemPath&
         // Top level item - Add New Prim (for all context op types).
         items.emplace_back(kUSDAddNewPrimItem, kUSDAddNewPrimLabel, Ufe::ContextItem::kHasChildren);
 
+        // Top level item - Bind/unbind existing materials
+        if (PXR_NS::UsdShadeMaterialBindingAPI::CanApply(fItem->prim())) {
+            // Show bind menu if there is at least one bindable material in the stage.
+            //
+            // TODO: Show only materials that are inside of the asset's namespace otherwise there
+            //       will be "refers to a path outside the scope" errors. See
+            //       https://groups.google.com/g/usd-interest/c/dmjV5bQBKIo/m/LeozZ3k6BAAJ
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/ufe/UsdContextOps.cpp:849]: Any idea how to do this?

File: lib/mayaUsd/ufe/UsdShaderNodeDef.cpp [hunk 2]
```diff
@@ -48,6 +48,9 @@ Ufe::ConstAttributeDefs getAttrs(const PXR_NS::SdrShaderNodeConstPtr& shaderNode
     for (const PXR_NS::TfToken& name : names) {
         PXR_NS::SdrShaderPropertyConstPtr property
             = input ? shaderNodeDef->GetShaderInput(name) : shaderNodeDef->GetShaderOutput(name);
+        if (!property) {
+            continue;
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/ufe/UsdShaderNodeDef.cpp:52]: I have third-party shaders crashing here. They registered the input names, but did not register a SdrShaderProperty.

File: lib/mayaUsd/ufe/UsdContextOps.cpp (hunk 1 above)
Comment (ppt-adsk) [reply lib/mayaUsd/ufe/UsdContextOps.cpp:849]: Afraid not.

File: lib/mayaUsd/ufe/UsdContextOps.cpp [hunk 3]
```diff
@@ -684,6 +840,44 @@ Ufe::ContextOps::Items UsdContextOps::getItems(const Ufe::ContextOps::ItemPath&
...
+            //       This might help restrict the stage traversal scope and improve performance.
+            //
+            // For completeness, and to point out that material assignments are complex:
+            //
+            // TODO: Introduce the "rendering purpose" concept
+            // TODO: Introduce material binding via collections API
+            //
+            auto stage = fItem->prim().GetStage();
+            for (UsdPrim currentPrim : stage->TraverseAll()) {
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdContextOps.cpp:858]: How expensive can this traversal end up being?

File: lib/mayaUsd/ufe/UsdShaderNodeDef.cpp (hunk 2 above)
Comment (ppt-adsk) [reply lib/mayaUsd/ufe/UsdShaderNodeDef.cpp:52]: That would be a great code comment.

**Review by ppt-adsk — APPROVED** [review]
Looks good to me!  Just minor comments.  Material discovery traversals are a bit worrisome, I will say, especially if this is used in a context where UI must be displayed synchronously, but I will assume that you feel the typical scene hierarchy where this will be used will not cause performance problems.

**Review by ppt-adsk — DISMISSED** [review]

**Review by feldstj — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**